bridge/github/export.go

```diff
@@ -0,0 +1,689 @@
+			}
+
+			hash, err = opr.Hash()
+			if err != nil {
+				return fmt.Errorf("comment hash: %v", err)
+			}
+
+		case *bug.SetStatusOperation:
+			opr := op.(*bug.SetStatusOperation)
+			if err := updateGithubIssueStatus(client, bugGithubID, opr.Status); err != nil {
+				return fmt.Errorf("updating status %v: %v", bugGithubID, err)
+			}
+
+			hash, err = opr.Hash()
+			if err != nil {
+				return fmt.Errorf("comment hash: %v", err)
+			}
+
+			id = bugGithubID
+			url = bugGithubURL
+
+		case *bug.SetTitleOperation:
+			opr := op.(*bug.SetTitleOperation)
+			if err := updateGithubIssueTitle(client, bugGithubID, opr.Title); err != nil {
+				return fmt.Errorf("editing comment %v: %v", bugGithubID, err)
+			}
+
+			hash, err = opr.Hash()
+			if err != nil {
+				return fmt.Errorf("comment hash: %v", err)
+			}
+
+			id = bugGithubID
+			url = bugGithubURL
+
+		case *bug.LabelChangeOperation:
+			opr := op.(*bug.LabelChangeOperation)
+			if err := ge.updateGithubIssueLabels(client, bugGithubID, opr.Added, opr.Removed); err != nil {
+				return fmt.Errorf("updating labels %v: %v", bugGithubID, err)
+			}
+
+			hash, err = opr.Hash()
+			if err != nil {
+				return fmt.Errorf("comment hash: %v", err)
+			}
+
+			id = bugGithubID
+			url = bugGithubURL
+
+		default:
+			panic("unhandled operation type case")
+		}
+
+		// mark operation as exported
+		if err := markOperationAsExported(b, hash, id, url); err != nil {
+			return fmt.Errorf("marking operation as exported: %v", err)
+		}
+	}
+
+	if err := b.CommitAsNeeded(); err != nil {
+		return fmt.Errorf("bug commit: %v", err)
+	}
+
+	return nil
+}
+
+// getRepositoryNodeID request github api v3 to get repository node id
+func getRepositoryNodeID(owner, project, token string) (string, error) {
+	url := fmt.Sprintf("%s/repos/%s/%s", githubV3Url, owner, project)
+
+	client := &http.Client{
+		Timeout: defaultTimeout,
+	}
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	// need the token for private repositories
+	req.Header.Set("Authorization", fmt.Sprintf("token %s", token))
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("error retrieving repository node id %v", resp.StatusCode)
+	}
+
+	aux := struct {
+		NodeID string `json:"node_id"`
+	}{}
+
+	data, _ := ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	err = json.Unmarshal(data, &aux)
+	if err != nil {
+		return "", err
+	}
+
+	return aux.NodeID, nil
+}
+
+func markOperationAsExported(b *cache.BugCache, target git.Hash, githubID, githubURL string) error {
+	_, err := b.SetMetadata(
+		target,
+		map[string]string{
+			keyGithubId:  githubID,
+			keyGithubUrl: githubURL,
+		},
+	)
+
+	return err
+}
+
+// get label from github
+func (ge *githubExporter) getGithubLabelID(label string) (string, error) {
+	url := fmt.Sprintf("%s/repos/%s/%s/labels/%s", githubV3Url, ge.conf[keyOwner], ge.conf[keyProject], label)
```

Comment:
```
  repository(owner: "MichaelMure", name: "git-bug") {
    label(name: "bug") {
      id
    }
    labels(first:10) {
      nodes{
        name
        id
      }
    }
  }
```

Reply:
^ you can either get all the IDs in one go or query them one by one when needed

Reply:
Haaaaaa !! x) will make changes to use this